rholang/src/main/k/rholang/tests/Std-Pattern-Matching/matching-with-the-new-construct.rho

```diff
@@ -0,0 +1,8 @@
+// A successful match rewrites to
+// Nil!("success")
+
+match
+  new x in { x!(Nil) }
+{
+  new y in { y!(z) } => { z!("success") }
+}
```

Comment:
This won't parse (although it probably should) because of how coercions are ordered.

Reply:
I'm not sure I understand why this doesn't parse?

Reply:
Well, when I tried running this contract it didn't pass the parser phase.

Reply:
Ah, there is a problem with this example. The `z` is a process variable, not a name variable, so `z!("success")` should be replaced with `@z!("success")`. In addition to that change, wrap both the target and the pattern in braces and that should run.

Reply:
Yes, exactly.

Reply:
You're absolutely right. They are working in the framework but I neglected to run them through the current rholang compiler. I'll do that for future examples.

Reply:
:+1: good work nevertheless.